src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +704,159 @@ public String[] executeSimple(Object[][] params) {
+				double second = (sorted[half]).doubleValue();
+				double median = (first+second)/2;
+				return new Number[]{median};
+			}
+		})
+			.description("Get the middle value of a sorted list of numbers. If the list has an even number of values, the median is the average of the two middle numbers.")
+			.examples(
+				"median(1, 2, 3, 4, 5) = 3",
+				"median(1, 2, 3, 4, 5, 6) = 3.5",
+				"median(0, 123, 456, 789) = 289.5"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<>("factorial", new Parameter[]{
+			new Parameter<>("number", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Number number = (Number) params[0][0];
+				Long total = Math2.factorial(number.longValue());
+				if (total == null)
+					return null;
+				return new Number[]{total};
+			}
+		})
+			.description()
+			.examples()
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("root", new Parameter[]{
+			new Parameter<>("root", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("number", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Number root = (Number) params[0][0];
+				if (root.intValue() == 0)
+					return null;
+				Number number = (Number) params[1][0];
+				return new Number[]{Math.pow(number.doubleValue(), (1 / root.doubleValue()))};
+			}
+		})
+			.description("Get the result of the 'number' being rooted to the power of 'root'")
+			.examples(
+				"root(2, 4) = 2 # same as sqrt(4)",
+				"root(4, 16) = 2",
+				"root(-4, 16) = 0.5"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("permutation", new Parameter[]{
+			new Parameter<>("total", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("set", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Number total = (Number) params[0][0];
+				Number set = (Number) params[1][0];
+				long denominator = total.longValue() - set.longValue();
+				Long totalF = Math2.factorial(total.longValue());
+				Long denominatorF = Math2.factorial(denominator);
+				if (totalF == null || denominatorF == null)
+					return null;
+				Skript.adminBroadcast("Num: " + totalF);
+				Skript.adminBroadcast("Den: " + denominatorF);
+				double result = totalF.doubleValue()/denominatorF.doubleValue();
+				return new Number[]{result};
+			}
+		})
+			.description(
+				"Get the number of possible ordered arrangements from numbers 1 to the provided 'total' within a 'set' size.",
+				"For example,  A permutation of 3 with a set size of 1 returns 3: (1), (2), (3)",
+				"A permutation of 3 with a set size of 2 returns 6: (1, 2), (1, 3), (2, 1), (2, 3), (3, 1), (3, 2)"
```

Comment:
```suggestion
				"A permutation with 3 options and an arrangement size of 2 returns 6: (1, 2), (1, 3), (2, 1), (2, 3), (3, 1), (3, 2)"
```
they dont return sets as `{2, 3} = {3, 2}`, so this would only return 3

Reply:
Thats combination